assets/js/googlesitekit/site/index.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { get, set } from 'googlesitekit-api';
+import {
+	FETCH_CONNECTION_INFO,
+	FETCH_RESET,
+	INITIALIZE,
+	RESET,
+	RESET_FAILURE,
+	RESET_SUCCESS,
+	RECEIVE_CONNECTION_INFO,
+	RECEIVE_CONNECTION_INFO_FAILED,
+} from './index.private';
+
+export const INITIAL_STATE = {
+	connectionInfo: null,
+	isFetchingConnectionInfo: false,
+	isResetting: false,
+};
+
+export const STORE_NAME = 'core/site';
+
+export const actions = {
+	fetchConnectionInfo() {
+		return {
+			payload: {},
+			type: FETCH_CONNECTION_INFO,
+		};
+	},
+
+	fetchReset() {
+		return {
+			payload: {},
+			type: FETCH_RESET,
+		};
+	},
+
+	initialize() {
+		return {
+			payload: {},
+			type: INITIALIZE,
+		};
+	},
+
+	receiveConnectionInfo( connectionInfo ) {
+		invariant( connectionInfo, 'connectionInfo is required.' );
+
+		return {
+			payload: { connectionInfo },
+			type: RECEIVE_CONNECTION_INFO,
+		};
+	},
+
+	receiveConnectionInfoFailed( error ) {
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { error },
+			type: RECEIVE_CONNECTION_INFO_FAILED,
+		};
+	},
+
+	*reset() {
+		try {
+			yield actions.fetchReset();
+			yield actions.resetSuccess();
+			yield actions.initialize();
```

Comment:
Not sure about the need for an `initialize` action at this point. I get where your thoughts come from, but I don't think that should be managed here, especially because the reset here only re-initializes this particularly store. We're also reloading the entire page anyway on reset, so this should be irrelevant. I guess what I'm saying is that this implementation is kind of half-baked so I'd prefer to not have it for now.

Reply:
> We're also reloading the entire page anyway on reset, so this should be irrelevant.

Are we? Why would we do that? If the entire app is managed by the store we could show a different UI on reset/(re-)initialisation.

That said, it still makes sense to zero out the store for testing here; we can't test the refresh behaviour here and I think it's worth ensuring the store behaves property. I suppose we could just test for `resetSuccess` being fired, but this feels more complete.

And I think most stores should have an `initialize()` action, at least for testing but they sometimes come in handy in real-world scenarios.

Reply:
That's fair. It may come in handy in the future, but I'd prefer to only introduce it once we see a need for it. There's not usage for it in tests yet, and for this reset it's rather irrelevant at the moment. It would be trivial to introduce this action for all stores once we have a need for this, so I'd say let's remove for now and revisit via separate issue once we have a bigger picture.